great_expectations/datasource/fluent/snowflake_datasource.py

```diff
@@ -468,13 +502,16 @@ def _build_engine_with_connect_args(
 
         engine_kwargs: dict[Literal["url", "connect_args"], Any] = {}
         if connect_args:
-            if connect_args.get("private_key"):
+            if private_key := connect_args.get("private_key"):
                 url_args.pop(  # TODO: update models + validation to handle this
                     "password", None
                 )
                 LOGGER.info(
                     "private_key detected, ignoring password and using private_key for authentication"
                 )
+                # assume the private_key is base64 encoded
+                connect_args["private_key"] = base64.standard_b64decode(private_key)
```

Comment:
non-blocking: might be good to not modify the dict that was passed in here.

Reply:
It makes no difference.
The dictionary comes from an internal `SnowflakeDatasource.dict()` call.
So, there's no risk of unintended mutation.